### PR TITLE
Pass around an interface so clients don't have access to anyth ing they shouldn't have.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/EditorFeatures/Core/Implementation/GoToDefinition/GoToDefinitionHelpers.cs [hunk 1]
```diff
@@ -68,7 +70,7 @@ internal static class GoToDefinitionHelpers
             var displayParts = NavigableItemFactory.GetSymbolDisplayTaggedParts(project, symbol);
             var title = displayParts.JoinText();
 
-            if (!preferredSourceLocations.Any())
+            if (preferredSourceLocations.Length == 0)
```
Comment (AdamSpeight2008) [inline src/EditorFeatures/Core/Implementation/GoToDefinition/GoToDefinitionHelpers.cs]: Would a switch be a better option ?
```c#
switch( preferredSourceLocations.Length )
{
case 0:
case 1:
default:
}

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
test this please

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
@jmarolf

**Pull request merged**